### ISSUE TITLE
Array validation in v3 - filtered out `null`s

### ISSUE THREAD

**Issue by angelikatyborska** [react-tools/react-form#257]
I am trying to upgrade from v2 to v3.

I used to write my validators so that for all values that are arrays, the errors are also arrays of the same length. If the value of the error at a corresponding index is null, that means the value in the form at that index is valid.

In v3, all null values in arrays of errors are removed and the array of errors is shorter than the array of values. This results in errors being wrongly assigned to fields. Here's a screenshot from my app where you can see an invalid value for the third item, yet the error is displayed for the first item:

![screen shot 2018-02-27 at 17 41 50](https://user-images.githubusercontent.com/7852553/36741491-8db31cda-1be5-11e8-86c9-54e97e6587c4.png)

Here's a very simple example that shows this problem https://codesandbox.io/s/r74rpj807q:
```js
<Form validate={() => ({ myArray: [null, 'oops', null] })}>
    {(formApi) => {
      return <div>
        <h1>My array:</h1>
        <div><Text field={['myArray', 0]} /></div>
        <div><Text field={['myArray', 1]} /></div>
        <div><Text field={['myArray', 2]} /></div>
        <h2>Errors:</h2>
        <pre>{JSON.stringify(formApi.getFormState().errors)}</pre>
      </div>
    }}
  </Form>
```

![screen shot 2018-02-27 at 17 36 21](https://user-images.githubusercontent.com/7852553/36741545-b0785320-1be5-11e8-81b1-55e29847d5bf.png)

Is this a bug or did the results expected from validators change?

**Comment by joepuzzo** [comment]
working on fixing this now

**Comment by joepuzzo** [comment]
Ok its fixed in V3.0.4 !
